engine/runtime/src/main/java/org/enso/interpreter/runtime/callable/atom/Atom.java

```diff
@@ -203,6 +205,18 @@ boolean hasFunctionalDispatch() {
     return true;
   }
 
+  @ExportMessage
+  @CompilerDirectives.TruffleBoundary
```

Comment:
Right now I am putting this call behind `TruffleBoundary` and using `InteropLibrary.getUncached()` - this is not going to be fast - it is possible to speed things up, if necessary.

Reply:
No more boundary - running at full speed at the end.